### PR TITLE
[1.x] Add Inertia select component

### PR DESCRIPTION
**Pull request by aarondfrancis**
Hey yall 👋 

I needed a select form input and wanted it to match the rest of jetstream, so I made a new component. Thought it might be nice to have in the library!

Here's a screenshot:

image

It uses the same SVG chevron as the profile dropdown.

Let me know if anything needs to be changed 🤙

### CODE REVIEW

**Comment by bonzai** [comment]
Is there any reason for not using the `form-select` class? It would make code much simpler:

```html
// no <div> wrapper
<select class="form-select rounded-md shadow-sm">
    <slot></slot>
</select>
```

Zrzut ekranu 2020-10-16 o 23 24 55

Also, you should make the Livewire version if you want your PR to be accepted.

**Comment by aarondfrancis** [comment]
Only that I totally forgot the form-select class existed! Updated.

Thanks @bonzai

File: stubs/inertia/resources/js/Jetstream/Select.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<template>
+    <select class="form-select w-full rounded-md shadow-sm" :value='value' @input="$emit('input', $event.target.value)" ref="select">
```
Comment (bonzai) [inline stubs/inertia/resources/js/Jetstream/Select.vue:2]:
```suggestion
    <select class="form-select rounded-md shadow-sm" :value="value" @input="$emit('input', $event.target.value)" ref="select">
```

File: stubs/inertia/resources/js/Jetstream/Select.vue [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <option v-if='placeholder' :value='null' disabled>{{ placeholder }}</option>
```
Comment (bonzai) [inline stubs/inertia/resources/js/Jetstream/Select.vue:3]:
```suggestion
        <option v-if="placeholder" :value="null" disabled>{{ placeholder }}</option>
```

Comment (bonzai) [reply stubs/inertia/resources/js/Jetstream/Select.vue:3]: I think `placeholder` prop is not needed - we can use `<slot></slot>` for that.

Comment (aarondfrancis) [reply stubs/inertia/resources/js/Jetstream/Select.vue:3]: IMO it's nice to have built in, and if you want to use the `<slot>` for that you still can, it doesn't prevent you in any way.

Comment (aarondfrancis) [reply stubs/inertia/resources/js/Jetstream/Select.vue:3]: Gah sorry, I thought StyleCI would catch all the quotes, etc.

**Comment by mous2890** [comment]
I love this idea of prebuilt components to use in Vue. Thank you.

**Comment by claudiodekker** [comment]
Hi, thanks for this, it's definitely something that will come in useful in some of my Inertia apps! 

That said, I'm a bit mixed on this being added to Jetstream, because to me it doesn't feel like it's the right place. My understanding is that Jetstream isn't a UI-component library but rather a UI-stack bootstrapping library.

Perhaps there's a different open source library that this would be more suitable in? Or am I missing something and this would also replace the existing selects in the Inertia Jetstream stack?

**Comment by aarondfrancis** [comment]
Yeah it's a good question! I don't know if I'm the person to answer it tbh. 

If Jetstream is a stack to bootstrap apps, then by necessity it will include some UI components, as it already does. The hard question is "how many UI components?" Probably only Taylor can answer that. 

From my point of view it seems reasonable to have a select form element to go along with inputs and buttons. 

I'd probably argue _against_ pulling Select2 in or something, as that is too much in my opinion. But having a native select with the right classes and a bit of dev UX seems right to me. 

All that to say... beats me! Just thought it was worth a shot.

**Comment by aarondfrancis** [comment]
Also @claudiodekker I must say that your response was extremely thoughtful and kind. You were basically saying "not sure this belongs here" but it was so nice and considerate. 

Thanks for that.

**Comment by taylorotwell** [comment]
I kind of agree that with @claudiodekker and I'm not sure I want to commit to maintaining components like this. However, I do really like this idea for an add-on package someone could build for Jetstream that has this component and other useful components like it.

**Comment by aarondfrancis** [comment]
Totally understood! Will assess my personal appetite for another package 🤔  😂

**Review by hugorodriguesbr — COMMENTED** [review]
Hi, tag select has no attribute value.

**Pull request closed without merging**